### ISSUE TITLE
Arrow key navigation in dropdowns / lists malfunctioning

### ISSUE THREAD

**Issue by scorster** [musescore/MuseScore#22179]
### Issue type

UX/Interaction bug (incorrect behaviour)

### Bug description

In long dropdowns and list views, arrow key navigation does not always function correctly:
- Pressing and holding the **Up arrow** works for a bit ... but eventually invokes a jump to the top of the list.
- The **Down arrow** does work properly when held (i.e while stillDown).

This can be reproduced for example in Properties > Text > Font face (requires a text element to be selected on the score); but in principle this applies to all dropdowns and list views.

### MuseScore Version

4.x

### Regression

Yes, this used to work in MuseScore 3.x and now is broken

### Operating system

Reported on macOS, probably applies to all

### Additional context

Originally reported at https://musescore.org/en/node/362510

Related: https://github.com/musescore/MuseScore/issues/16508 and [#24513](https://github.com/musescore/MuseScore/issues/24513) (a yet-to-be-created issue about pageUp/pageDown)

**Comment by zacjansheski** [comment]
Same on Windows

**Comment by scorster** [comment]
Just realized this issue is more widespread than I realized

In MS4:

• make sure no notes are selected
• select header text like Title Text, Subtitle, Composer ...
• in Text>Font press a letter **a** thru **g**

**Focus related errant result:** As reported initially Musescore sets the first note of the score to the keypress, but I thought that only happened when a Text Object was selected, and where the result affected the pitch of the parent note.

In MS4:

• make sure no notes are selected
• select header text like Title Text, Subtitle, Composer ...
• in Text>Font press **r**
  
![No Note or Rest selected - MS4 alert](https://github.com/user-attachments/assets/692875f2-b421-4b0c-8839-2b0191648e42)

**Comment by scorster** [comment]
Issues extant in MuseScore 4.4.1

While readying a bug report on a separate issue I wanted to set a font face for a Text field. And that once again put me face to face with these font menu bugs. I'm guessing QT is handing the menus improperly and that's difficult to circumvent which makes bugs like these linger for so long. I fear that issues with such basic matters likely mar MuseScore image. So it would be great to see this move out of the Backlog.

**Comment by cbjeukendrup** [comment]
The issues with letter keys are actually covered by https://github.com/musescore/MuseScore/issues/16508. Let's keep this issue focused on the problem with the arrow key, which is absolutely not limited to the font drop-down: other dropdowns, but also long lists like in the Instruments dialog show similar problems.

**Comment by scorster** [comment]
@cbjeukendrup Thanks for you reply.

Are you suggesting—with respect to menu navigation in general—I should open a separate Issue about the problems observed with pageUp and pageDown keys?

**Comment by cbjeukendrup** [comment]
Ah, sorry, I had missed the bit about pageUp and pageDown initially. Since this will most likely require a separate fix, it would indeed be good to open a separate issue, perhaps with a reference to this issue in "Additional context".

**Comment by cbjeukendrup** [comment]
I simplified the issue description a bit; I hope you don't mind. After creating the pageUp/pageDown issue, you can add a link to it to the bottom of this issue's description.

**Comment by worldwideweary** [comment]
Don't know if there is a duplicate post about this related to numerical boxes, but it's been like this in MSS4 since it broke off from the 3.x series
Like so: 

https://github.com/user-attachments/assets/3209d1f0-a075-46ea-834e-499a3a46887b

The arrows just don't do anything unlike standard convention.

Always seemed rather odd, but I suppose they wanted to break away from standard convention or just omitted it out of priority concerns or some sort of thing (kind of like the way the mouse doesn't work like normal applications with drop down menus, and never was updated: https://github.com/musescore/MuseScore/issues/28240)
